File: onyx-database-tests/src/main/java/entities/InheritedAttributeEntity.java
package entities;

import com.onyx.persistence.IManagedEntity;
import com.onyx.persistence.annotations.Attribute;
import com.onyx.persistence.annotations.Entity;
import com.onyx.persistence.annotations.Identifier;

/**
 * Created by timothy.osborn on 11/3/14.
 */
@Entity
public class InheritedAttributeEntity extends AbstractInheritedAttributes implements IManagedEntity
{
    @Identifier
    @Attribute
    public String id;

}

File: onyx-database-tests/src/main/kotlin/entities/InheritedAttributeEntity.kt
package entities

import com.onyx.persistence.IManagedEntity
import com.onyx.persistence.annotations.Attribute
import com.onyx.persistence.annotations.Entity
import com.onyx.persistence.annotations.Identifier

/**
 * Created by timothy.osborn on 11/3/14.
 */
@Entity
class InheritedAttributeEntity : AbstractInheritedAttributes(), IManagedEntity {
    @Identifier
    @Attribute
    var id: String? = null

}
